Userland/Services/WindowServer/ConnectionFromClient.cpp

```diff
@@ -804,6 +804,33 @@ Messages::WindowServer::GetSystemThemeResponse ConnectionFromClient::get_system_
     return name;
 }
 
+Messages::WindowServer::SetSystemThemeOverrideResponse ConnectionFromClient::set_system_theme_override(Core::AnonymousBuffer const& theme_override)
+{
+    bool success = WindowManager::the().set_theme_override(theme_override);
+    return success;
+}
+
+Messages::WindowServer::GetSystemThemeOverrideResponse ConnectionFromClient::get_system_theme_override()
+{
+    auto theme_override = WindowManager::the().get_theme_override();
+    if (!theme_override.has_value()) {
+        did_misbehave("GetSystemThemeOverrideResponse: Theme not overridden");
+        return nullptr;
+    }
+
+    return *theme_override;
+}
+
+void ConnectionFromClient::clear_system_theme_override()
+{
+    WindowManager::the().clear_theme_override();
+}
+
+Messages::WindowServer::IsSystemThemeOverriddenResponse ConnectionFromClient::is_system_theme_overridden()
```

Comment:
This creates a rather unfortunate, albeit unlikely TOCTOU situation.

Reply:
we can keep it for now, just pointing it out :)
ideally IPC would allow us to return an `Optional<T>`, I don't know if it does?

Reply:
`get_system_theme_override` returns an optional now